R/teal_lockfile.R

```diff
@@ -0,0 +1,120 @@
+#' Generate lockfile for application reproducibility
+#'
+#' This function is invoked during [teal::init] to create `renv`-compatible lockfile for use within the application.
```

Comment:
> For more information about lockfile creation visit [`teal_lockfile()`].

And it redirects here where creation is explained along with `future` process. Part about the `future` process doesn't make it easier to comprehend. Can we skip these technical details and maybe replace that lockfile is computed in a background when app is starting (the rest can be concluded from the code)

Reply:
The technical details are just in `@note` :)! If in this case this is too much, or confusing, or doesn't help future devs to understand what's going on than I'm OK removing that. What do you think @pawelru ?

Reply:
I think that the roxygen docs is mainly for users and I agree that it is probably too technical. If you want to add helpers for devs that let's do that in the code-comments.

Reply:
Alrighty, extended comments a bit and removed the note with details of `future` asynchronous computations 072a989807acc8e8327df15bc1470afa679166a4